runelite-client/src/main/java/net/runelite/client/plugins/bosslogger/LootRecordPanel.java

```diff
@@ -0,0 +1,162 @@
+	{
+		this.record = record;
+
+		GridBagLayout layout = new GridBagLayout();
+		this.setLayout(layout);
+		this.setBorder(new MatteBorder( 0, 0, 1, 0, Color.GRAY));
+
+		// Item Image Icon
+		JLabel icon = new JLabel();
+		this.record.getIcon().addTo(icon);
+		// Item Name (Colored off Item Price)
+		JLabel item_name = new JLabel(this.record.getItem_name());
+		ColorLabel(item_name, this.record.getValue());
+		// Item Values (Colored off Total Value of item)
+		JLabel total = new JLabel(numberToString(this.record.getTotal()), SwingConstants.RIGHT);
+		ColorLabel(total, this.record.getTotal());
+
+		GridBagConstraints c = new GridBagConstraints();
+		c.fill = GridBagConstraints.BOTH;
+		c.weightx = 1;
+		c.gridx = 0;
+		c.gridy = 0;
+		c.ipady = 20;
+
+		// Add to Panel
+		this.add(icon, c);
+		c.gridx++;
+		this.add(item_name, c);
+		c.gridx++;
+		this.add(total, c);
+	}
+
+	// Used specifically for the Total Value element inside the tab
+	LootRecordPanel(Long totalValue)
+	{
+		GridBagLayout layout = new GridBagLayout();
+		this.setLayout(layout);
+		this.setBorder(new MatteBorder( 0, 0, 1, 0, Color.GRAY));
+
+		GridBagConstraints c = new GridBagConstraints();
+		c.fill = GridBagConstraints.BOTH;
+		c.weightx = 1;
+		c.gridx = 0;
+		c.gridy = 0;
+		c.ipady = 20;
+
+		JLabel totalText = new JLabel("Total Value:", SwingConstants.LEFT);
+		ColorLabel(totalText, totalValue);
+
+		// Item Values (Colored off Total Value of item)
+		JLabel total = new JLabel(numberToString(totalValue), SwingConstants.RIGHT);
+		ColorLabel(total, totalValue);
+
+		this.add(totalText, c);
+		c.gridx++;
+		this.add(total, c);
+	}
+
+
+	// Used specifically for the Killcount entry
+	LootRecordPanel(int size, int last)
+	{
+		GridBagLayout layout = new GridBagLayout();
+		this.setLayout(layout);
+		this.setBorder(new MatteBorder( 0, 0, 1, 0, Color.GRAY));
+
+		GridBagConstraints c = new GridBagConstraints();
+		c.fill = GridBagConstraints.BOTH;
+		c.weightx = 1;
+		c.gridx = 0;
+		c.gridy = 0;
+		c.ipady = 20;
+
+
+		JLabel current = new JLabel("Current KC: " + last, SwingConstants.LEFT);
+		current.setForeground(Color.CYAN);
+		JLabel recorder = new JLabel("Kills Logged: " + size, SwingConstants.RIGHT);
+		recorder.setForeground(Color.CYAN);
+
+		this.add(current, c);
+		c.gridx++;
+		this.add(recorder, c);
+	}
+
+	// Add K and M Suffix to items
+	private String numberToString(long v)
+	{
+		String suffix = " gp";
+		if (v >= 10000000)
+		{
+			v = v / 1000000;
+			suffix = "M" + suffix;
+
+		}
+		else if (v >= 100000)
+		{
+			v = v / 1000;
+			suffix = "K" + suffix;
+		}
+
+		return (new DecimalFormat("#,###.#").format(v) + suffix);
+	}
+
+	// Color label to match RuneScape coloring
+	private void ColorLabel(JLabel label, long val)
+	{
+		Color labelColor = Color.yellow;
+		if (val >= 10000000)
+		{
+			labelColor = Color.green;
+		}
+		else if (val >= 100000)
+		{
+			labelColor = Color.white;
+		}
+		label.setForeground(labelColor);
```

Comment:
Also this methodcould be reduced by a bunch of lines by using a ternary operator.